src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -1,6 +1,11 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+// Rare enums types not expressible in C# are not supported in native AOT
+#if !NATIVEAOT
+#define RARE_ENUMS
```

Comment:
@SamMonoRT @kotlarmilos should this also include all mobile/web targets?

Reply:
Yes, if we can include the mobile/web targets to mitigate the app size regressions, we should include those now, till we are able to identify and fix the Mono AOT compiler limitations.

Reply:
I have pushed a commit to disable the rare enums support for Mono as well. Please take a look.

Reply:
I've inspected native symbols of an iOS app and non-integral types except `bool` are supported in the Mono AOT engine.

Reply:
Ok, I am going to keep the rare enums enabled for Mono. I do not data to make the call on whether to disable it for Mono or not. It is two line change to enable/disable this for given config.

Reply:
Sounds good, leave it enabled for now. We can run some measurements and see if it helps in size if we disable it.

Reply:
It doesn't look to me like breaking change - by doing that we will restrict some theoretical or borderline cases. I would suggest excluding them as they are not expressible in C# and such change could mitigate the size regression. Some edge cases might fail where non-integral enums are created by using reflection, which is not recommended by the documentation. 

Let's hear the other opinions as well. If it is a blocker, we can proceed and update it once we align.